File: Lv.1/추억 점수.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    map<string,int> m;
    
    // map에 추가
    for(int i=0;i<name.size();i++){
        
        m[name[i]]=yearning[i];
        
    }
    
    // 각 사진별 점수 계산
    for(int i=0;i<photo.size();i++){
        int temp = 0;
        for(int j=0;j<photo[i].size();j++){
            if(m.find(photo[i][j])!=m.end()){
                temp+=m[photo[i][j]];
            }
            else{
                continue;
            }
        }
        answer.push_back(temp);
    }
    return answer;
}

File: Java/LV1/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        HashMap<String,Integer> hm = new HashMap<>();
        
        for(int i=0;i<name.length;i++){
            String curName = name[i];
            int score = yearning[i];
            
            hm.put(curName,score);
        }
        
        for(int i=0;i<photo.length;i++){
            
            int sum = 0;
            for(int j=0;j<photo[i].length;j++){
                String curName = photo[i][j];
                if(hm.containsKey(curName)){
                    sum+=hm.get(curName);
                }
            }
            answer[i]=sum;
        }
        
//         for(Map.Entry<String,Integer> k : hm.entrySet()){
//             String key = k.getKey();
//             Integer value = k.getValue();
            
//             System.out.println(key+' '+value);
//         }
        
        return answer;
    }
}
